src/components/modal/Modal.js

```diff
@@ -15,6 +15,8 @@ type Props = {
     children: React.Node,
     /** Additional CSS classname of the `.modal` element */
     className?: string,
+    /** If provided, will render a button with this text instead of the 'X' to dismiss the modal (onRequestClose must also be provided) */
+    closeButtonText?: string,
```

Comment:
This seems fine, but at minimum i would change to React.Node instead of string.

Can you foresee something different being there besides an x or a white button with a label? Instead of passing in a string, what about passing in a render prop? e.g. `renderCloseButton: (closeButtonProps) => React.Node`

Reply:
+2